### PR TITLE
[sensors]Cancel sensors example timer avoid exception

### PR DESCRIPTION
**Pull request by includecmath**
## Description

*I did some automation tools for plugins, and find sensors example may have setState() called after dispose() exception.*

## Related Issues

[VERBOSE-2:ui_dart_state.cc(157)] Unhandled Exception: setState() called after dispose(): SnakeState#8dc20(lifecycle state: defunct, not mounted)
This error happens if you call setState() on a State object for a widget that no longer appears in the widget tree (e.g., whose parent widget no longer includes the widget in its build). This error can occur when code calls setState() from a timer or an animation callback.
The preferred solution is to cancel the timer or stop listening to the animation in the dispose() callback. Another solution is to check the "mounted" property of this object before calling setState() to ensure the object is still in the tree.
This error might indicate a memory leak if setState() is being called because another object is retaining a reference to this State object after it has been removed from the tree. To avoid memory leaks, consider breaking the reference to this object during dispose().
#0      State.setState.<anonymous closure> (package:flutter/src/widgets/framework.d<…>

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by includecmath** [comment]
CI seems time out. How to re-run it?

**Comment by includecmath** [comment]
Hi @ened, @mklim CI publishable warning check failed? pubspec.yaml fields in Flutter 1.10.0 seems incompatible?
Also this file no default reviewers.

**Comment by ened** [comment]
Hi @includecmath , could you please take a look at this error message:

```
Suggestions:
* Your pubspec.yaml includes an "author" section which is no longer used and may be removed.
* In pubspec.yaml the flutter.plugin.{androidPackage,iosPrefix,pluginClass} keys are deprecated. Consider using the flutter.plugin.platforms key introduced in Flutter 1.10.0
  See https://flutter.dev/docs/development/packages-and-plugins/developing-packages#plugin
```

I haven't done it myself, yet but it seems we need to switch the plugin to the federated pubspec.yaml style: https://flutter.dev/docs/development/packages-and-plugins/developing-packages#plugin-platforms.

**Comment by includecmath** [comment]
@ened If publishable check use Flutter 1.10.0 new pubspec.yaml style, Can I write a script and modify all plugins uniformly.

**Comment by includecmath** [comment]
@ened Many files seems no default reviewers.

**Comment by ened** [comment]
Hi @includecmath the plugins repository has seen some significant updates for all pubspec files and is now building again nicely, so we can continue on the regular review for PRs.
Regarding this one, could you please check if we  need to add a `StreamSubscription<AccelerometerEvent>` field and bind it to the call made in `initState`?
Thank you

**Comment by includecmath** [comment]
Hi @ened This is a good suggestion and code specification cancel the StreamSubscription.

**Comment by collinjackson** [comment]
@ened is this ready to merge from your perspective?

**Comment by ened** [comment]
@collinjackson yes, ready to go. Personally, I would sort the methods in a stateful widget like so:

```
initState
build
dispose
_*
```

But we don't have a style guide for that, therefore LGTM!

File: packages/sensors/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.4.1+5
+
+* Fix example setState() called after dispose() by canceling the timer.
```
Comment (collinjackson) [inline packages/sensors/CHANGELOG.md]:
```suggestion
* Fix example `setState()` called after `dispose()` by canceling the timer.
```

**Pull request merged**